src/Controller/UsersController.php

```diff
@@ -437,24 +437,69 @@ public function deleteUser(Request $request, PDO $db)
      * @throws Exception
      */
     public function setTrusted(Request $request, PDO $db)
+    {
+        $this->setAdminOnlyBooleanField($request, $db, 'trusted');
+    }
+
+    /**
+     * Allow users to be set as admin
+     *
+     * @param $request Request
+     * @param $db      PDO
+     *
+     * @throws Exception
+     */
+    public function setAdmin(Request $request, PDO $db)
+    {
+        $this->setAdminOnlyBooleanField($request, $db, 'admin');
+    }
+
+    /**
+     * Handles setting a boolean value that only an admin can set, since we have the same pattern happening 2x
+     * (and may add setting for the activity bool as well)
+     *
+     * @param $request Request
+     * @param $db      PDO
+     * @param $field   string
+     *
+     * @throws Exception
+     */
+    protected function setAdminOnlyBooleanField(Request $request, PDO $db, $field)
     {
         if (false == ($request->getUserId())) {
             throw new Exception("You must be logged in to change a user account", 401);
         }
 
+        if (!in_array($field, ['trusted', 'admin'])) {
```

Comment:
If we use `true` as a third variable for `in_array`, then we can be sure about the types here.